### PR TITLE
Add support for ignoring VCS directories

### PR DESCRIPTION
**Pull request by jderehag**
Related to PR #69 but works for both vcs backends (git and hg) and also supports globbed paths.

Signed-off-by: Jesper Derehag <jderehag@hotmail.com>

### CODE REVIEW

File: fish_prompt.fish [hunk 1]
```diff
@@ -92,8 +92,19 @@ function __bobthefish_pretty_parent -S -a current_dir -d 'Print a parent directo
   string replace -ar '(\.?[^/]{'"$fish_prompt_pwd_dir_length"'})[^/]*/' '$1/' "$parent_dir/"
 end
 
+function __ignore_vcs_dir -d 'Identifies if dir is to be ignored for VCS highlighting'
+  for path_ in $vcs_ignore_paths
+    if [ (string match -r (realpath $path_ ^/dev/null) (realpath $PWD ^/dev/null)) ]
```
Comment (bobthecow) [inline fish_prompt.fish]: this probably should be a regular match (i.e. glob) rather than a regex match.

File: fish_prompt.fish [hunk 2]
```diff
@@ -92,8 +92,19 @@ function __bobthefish_pretty_parent -S -a current_dir -d 'Print a parent directo
   string replace -ar '(\.?[^/]{'"$fish_prompt_pwd_dir_length"'})[^/]*/' '$1/' "$parent_dir/"
 end
 
+function __ignore_vcs_dir -d 'Identifies if dir is to be ignored for VCS highlighting'
+  for path_ in $vcs_ignore_paths
```
Comment (bobthecow) [inline fish_prompt.fish]: we try to use a `theme_` prefix for as many theme related config options as possible. should this use that prefix as well?

File: fish_prompt.fish (hunk 1 above)
Comment (jderehag) [reply fish_prompt.fish]: doh, of course you are right... I'll fix it..

File: fish_prompt.fish (hunk 2 above)
Comment (jderehag) [reply fish_prompt.fish]: I suspected as much but wasnt sure, I will rename it..

**Comment by jderehag** [comment]
@bobthecow Ok, updated according to comments and also added some notes in the readme.

**Comment by bobthecow** [comment]
Thanks, merged!

After playing with this for a bit, I decided it needed an implicit `/*` on the end of every pattern. Otherwise, the repo root would not have a vcs segment in the prompt, but subdirectories inside the repo would show it, and that's super weird :)

**Pull request merged**